engine/access/ingestion/engine.go

```diff
@@ -427,33 +432,35 @@ func (e *Engine) handleExecutionReceipt(originID flow.Identifier, r *flow.Execut
 		return fmt.Errorf("failed to store execution receipt: %w", err)
 	}
 
-	block, err := e.blocks.ByID(r.ExecutionResult.BlockID)
-	if err != nil {
-		return fmt.Errorf("failed to lookup block while handling receipt: %w", err)
-	}
-
-	if block.Header.Height > e.maxReceiptHeight {
-		e.transactionMetrics.UpdateExecutionReceiptMaxHeight(block.Header.Height)
-		e.maxReceiptHeight = block.Header.Height
-	}
-
-	e.trackExecutedMetricForReceipt(r)
+	e.trackExecutionReceiptMetrics(r)
 	return nil
 }
 
-func (e *Engine) trackExecutedMetricForReceipt(r *flow.ExecutionReceipt) {
+func (e *Engine) trackExecutionReceiptMetrics(r *flow.ExecutionReceipt) {
 	// TODO add actual execution time to execution receipt?
 	now := time.Now().UTC()
 
 	// retrieve the block
 	b, err := e.blocks.ByID(r.ExecutionResult.BlockID)
+
 	if errors.Is(err, storage.ErrNotFound) {
 		e.blocksToMarkExecuted.Add(r.ExecutionResult.BlockID, now)
+
+		// execution receipts are received as they are sent by ENs, which may be before the block
+		// is finalized and processed through the sync engine. When that happens, just use the
+		// latest finalized height.
+		e.transactionMetrics.UpdateExecutionReceiptMaxHeight(e.highestFinalizedHeight)
```

Comment:
do we want to report this "wrong" statement. 

A finalized block doesn't mean it's executed. A sealed block is executed, but not finalized.

Reply:
I updated it to just update the metric when the block is finalized